### ISSUE TITLE
OKD and Fedora core os

### ISSUE THREAD

**Issue by TheFrisianClause** [coreos/fedora-coreos-tracker#709]
**Describe the bug**
When I try to install an OKD Cluster with the usage of Craig Robinsons tutorial, I get alot of 'connection refused' errors when the bootstrapping is in progress. When I deploy all of the nodes, Bootstrap, 3 masters and 2 worker nodes and just let it run the bootstrapping, it will fail. But after this has been done I can change the SELinux settings to 'permissive' mode via ssh by using the 'core' user. Then rerunning the bootstrap process and nothing fails and just continues without any issues.

Though I do run into some issues later on again, because apparently something hasn't be done correctly while the bootstrap process has been running and because I needed to reboot the bootstrap node and all the other nodes to have the SELinux permissive mode take effect.

**Reproduction steps**
1.  Follow Craig Robinsons tutorial on OKD (Link: https://medium.com/@craig_robinson/guide-installing-an-okd-4-5-cluster-508a2631cbee)
2. Deploy all the nodes
3.  Start the nodes one by one and let it run.
4. When everything is done and you have alot of 'connection/connection refused' errors, SSH into each and every machine to set SELinux to permissive mode.
5. Rerun the bootstrapping process.
6. Now it completes without issues.

**Expected behavior**
No connection errors, when running bootstrapping. 

**Actual behavior**
Connection errors due to SELinux

**System details**
 - Bare Metal - VMware ESXi 7.0U1C
 - Fedora CoreOS version - 33.20201214.3.1 stable

**Additional information**
Is there any way to set some kind of kernelboot parameters to set SELinux to 'permissive' mode on boot? Instead of have to run the whole bootstrapping proces and then eventually still fail the OKD Cluster installation?

**Comment by jlebon** [comment]
Do you have the logs of the SELinux denials? (Try `journalctl --grep 'avc.*denied'`.)
You can boot with `enforcing=0` to put SELinux in permissive mode.
/cc @LorbusChris @vrutkovs

**Comment by JulianFRL** [comment]
Here is a screenshot of it, I have done a fresh install of the bootstrap node.
![image](https://user-images.githubusercontent.com/22170757/104029583-7306c800-51ca-11eb-8238-23a4e6227c55.png)

**Comment by vrutkovs** [comment]
SELinux denial bug is https://bugzilla.redhat.com/show_bug.cgi?id=1896796, however the real problem here is `dial: tcp: lookup quay.io on [::1]:53` - network/DNS is misconfigured

**Comment by JulianFRL** [comment]
I am currently trying to solve the 'dial tcp' error, as I might have a solution for that but I am not completely sure.

**Comment by jlebon** [comment]
> SELinux denial bug is [bugzilla.redhat.com/show_bug.cgi?id=1896796](https://bugzilla.redhat.com/show_bug.cgi?id=1896796), however the real problem here is `dial: tcp: lookup quay.io on [::1]:53` - network/DNS is misconfigured

Hmm, I think it's the same issue: SELinux is breaking systemd-resolved, which then breaks DNS.

This should be fixed in the latest release though (that RHBZ has "Fixed In Version: selinux-policy-3.14.6-33.fc33", which is in 33.20201214.3.1).

@JulianFRL Can you confirm that the boot media itself is that version as well?

**Comment by JulianFRL** [comment]
Yes I used this .iso file from the Fedora Core OS site: fedora-coreos-33.20201214.3.1-live.x86_64.iso
I don't know if there are any other iso files?

Currently I am testing if this might solve this issue, by building a DNS server in the same network as the OKD network 192.168.4.x, as I have a DNS server in a different network 192.168.3.x.
I pointed the .4.x VLAN DHCP DNS settings to the 192.168.3.x DNS server.

**Comment by vrutkovs** [comment]
Looks like a dupe of https://github.com/openshift/okd/issues/380: use [older F32 stable release](https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/32.20201104.3.0/x86_64/fedora-coreos-32.20201104.3.0-live.x86_64.iso)

**Comment by JulianFRL** [comment]
Will try that thank you :)

**Comment by JulianFRL** [comment]
Well after a whole weekend of troubleshooting I finally found the right match and got it working.
The Fedora Core OS Download which @vrutkovs mentioned above me works like a charm in combination with OKD 4.5.0-0.okd-2020-10-15-235428

I tested it with the latest OKD 4.6.0 as well but I had alot of troubles with it and it just wouldnt seem to work for some reason (Maybe I forgot something I don't know). With this version OKD 4.5.0-0.okd-2020-10-15-235428 I almost had to do nothing but wait 5 - 10 more minutes and then everything was deployed.
